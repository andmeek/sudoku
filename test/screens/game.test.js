import { shallowMount } from '@vue/test-utils'
import Game from '../../src/screens/game.vue'

jest.useFakeTimers()

describe('Game.vue', () => {
  let wrapper, board

  beforeEach(() => {
    board = genTestBoard()
    wrapper = shallowMount(Game, {
      propsData: {
        difficulty: 'Easy',
        board: board
      }
    })
  })

  test('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('emits a gamecompleted when the game is completed and the user clicks the okay button', () => {
    board.tiles.forEach((tile) => {
      tile.userValue = tile.actualValue
    })

    wrapper.vm.$forceUpdate()
    expect(wrapper.find('.completed')).toBeTruthy()

    wrapper.findAll('button').at(0).trigger('click')
    expect(wrapper.emitted().gamecompleted).toBeTruthy()
  })

  test('registers and removes a timer as part of the component lifecycle', () => {
    expect(setInterval).toHaveBeenCalledTimes(1)
    expect(setInterval).toHaveBeenCalledWith(expect.any(Function), 1000)
    expect(clearInterval).toHaveBeenCalledTimes(0)

    jest.advanceTimersByTime(3000)

    expect(board.timer).toEqual(3)

    wrapper.destroy()

    jest.advanceTimersByTime(1000)
    expect(clearInterval).toHaveBeenCalledTimes(1)
    expect(board.timer).toEqual(3)
  })
})
