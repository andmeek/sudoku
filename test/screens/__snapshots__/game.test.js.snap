// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game.vue matches snapshot 1`] = `
<div class="game screen">
  <game-header-stub difficulty="Easy" board="[object Object]"></game-header-stub>
  <game-board-stub board="[object Object]"></game-board-stub>
  <user-input-stub board="[object Object]"></user-input-stub>
  <!---->
</div>
`;

exports[`Game.vue when the board is completed matches snapshot 1`] = `
<div class="game screen">
  <!---->
  <!---->
  <!---->
  <div class="completed">
    You win!

    <button>Okay</button></div>
</div>
`;
